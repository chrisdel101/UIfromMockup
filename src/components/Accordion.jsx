import React from 'react'
import Icon from './Icon'
import { Accordion, Card } from 'react-bootstrap'

function myAccordion(props) {
  return (
    <Accordion defaultActiveKey="0">
      <Card>
        <Accordion.Toggle as={Card.Header} eventKey="0">
          <span className="accordion-icon">
            <Icon type={props.drawers['1'].icon} />
          </span>
          {props.drawers['1'].heading}
          <span className="accordion-arrow">
            <Icon type={props.drawers['1'].arrows[0]} />
          </span>
        </Accordion.Toggle>
        <Accordion.Collapse eventKey="0">
          <Card.Body>{props.drawers['1'].text}</Card.Body>
        </Accordion.Collapse>
      </Card>
      <Card>
        <Accordion.Toggle as={Card.Header} eventKey="1">
          <span className="accordion-icon">
            <Icon type={props.drawers['2'].icon} />
          </span>
          {props.drawers['2'].heading}
          <span className="accordion-arrow">
            <Icon type={props.drawers['1'].arrows[0]} />
          </span>
        </Accordion.Toggle>
        <Accordion.Collapse eventKey="1">
          <Card.Body>{props.drawers['2'].text}</Card.Body>
        </Accordion.Collapse>
      </Card>
      <Card>
        <Accordion.Toggle as={Card.Header} eventKey="1">
          <span className="accordion-icon">
            <Icon type={props.drawers['3'].icon} />
          </span>
          {props.drawers['3'].heading}
          <span className="accordion-arrow">
            <Icon type={props.drawers['1'].arrows[0]} />
          </span>
        </Accordion.Toggle>
        <Accordion.Collapse eventKey="1">
          <Card.Body>{props.drawers['3'].text}</Card.Body>
        </Accordion.Collapse>
      </Card>
    </Accordion>
  )
}

export default myAccordion
