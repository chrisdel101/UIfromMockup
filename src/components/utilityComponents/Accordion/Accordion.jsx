import React from 'react'
import { makeStyles } from '@material-ui/core/styles'
import ExpansionPanel from '@material-ui/core/ExpansionPanel'
import ExpansionPanelDetails from '@material-ui/core/ExpansionPanelDetails'
import ExpansionPanelSummary from '@material-ui/core/ExpansionPanelSummary'
import Typography from '@material-ui/core/Typography'
import ExpandMoreIcon from '@material-ui/icons/ExpandMore'
import Icon from '../Icon/Icon'
import './index.css'

// import './index.css'

const useStyles = makeStyles(theme => ({
  root: {
    width: '100%'
  },
  heading: {
    fontSize: theme.typography.pxToRem(15),
    flexBasis: '33.33%',
    flexShrink: 0
  },
  secondaryHeading: {
    fontSize: theme.typography.pxToRem(15),
    color: theme.palette.text.secondary
  }
}))

function ControlledExpansionPanels(props) {
  const classes = useStyles()
  const [expanded, setExpanded] = React.useState(false)

  const handleChange = panel => (event, isExpanded) => {
    setExpanded(isExpanded ? panel : false)
  }
  // get first panel open on load
  function setAttributes(i) {
    return [
      { defaultExpanded: true },
      { expanded: expanded === `panel2` },
      { expanded: expanded === `panel3` }
    ]
  }

  return (
    <div className={`${classes.root} accordion`}>
      {Object.values(props.drawers).map((drawer, i) => {
        return (
          <ExpansionPanel
            key={i}
            {...setAttributes(i)[i]}
            onChange={handleChange(`panel${i + 1}`)}
            className="expansion-panel"
          >
            <ExpansionPanelSummary
              className="expansion-panel-inner"
              expandIcon={<ExpandMoreIcon />}
              aria-controls={`panel${i + 1}bh-content`}
              id={`panel${i + 1}bh-header`}
            >
              <Icon type={drawer.icon} i={i} />
              <Typography className={`${classes.heading} panel-heading`}>
                {' '}
                {drawer.heading}
              </Typography>
              <Typography className={classes.secondaryHeading}>
                I am an expansion panel
              </Typography>
            </ExpansionPanelSummary>
            <ExpansionPanelDetails>
              <Typography>{drawer.text}</Typography>
            </ExpansionPanelDetails>
          </ExpansionPanel>
        )
      })}
    </div>
  )
}

export default ControlledExpansionPanels
