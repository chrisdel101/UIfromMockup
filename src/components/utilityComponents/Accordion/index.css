.expansion-panel .expansion-panel-inner {
  min-height: 75px;
  margin: 10px 0;
}
.expansion-panel .expansion-panel-inner.Mui-expanded {
  border: 1px solid #e5e5e5;
  position: relative;
  bottom: 10px;
}
.expansion-panel.MuiPaper-elevation1 {
  margin: 10px 0;
  border: 1px solid #e5e5e5;
  box-shadow: initial;
}
.main-accordion-container .panel-heading {
  font-family: 'Montserrat', sans-serif;
  text-transform: uppercase;
}
.main-accordion-container .icon-container {
  color: #95e1d3;
  font-size: 25px;
}
.main-accordion-container ::-webkit-scrollbar-thumb {
  background-color: #95e1d3;
}
.main-accordion-container {
  margin: 40px 0;
  display: grid;
  grid-template-columns: repeat(2, 1fr);
  min-height: 430px;
}
.main-accordion-container .image-container img {
  width: 100%;
  height: 100%;
}

.MuiSvgIcon-root {
  font-size: 2.25em;
  color: #d0d0d0;
}
.main-accordion-container ::-webkit-scrollbar-track {
  box-shadow: inset 0 0 5px grey;
  border-radius: 10px;
  background: #95e1d3;
}
.main-accordion-container .MuiCollapse-container {
  overflow-y: scroll;
  max-height: 100px;
}
