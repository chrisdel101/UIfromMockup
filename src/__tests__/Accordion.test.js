import React from 'react'
import Accordion from '../components/utilityComponents/Accordion/Accordion.jsx'
import renderer from 'react-test-renderer'

test('create and match snapshot', () => {
  const drawers = {
    '1': {
      icon: 'picture-o',
      heading: 'photography',
      text:
        'Nor whom love fountain seemed ive chaste did. Heal dote into take their native begun. Of he can way loved cared him had to dear, from him parting flaunting noontide.'
    },
    '2': {
      icon: 'sliders',
      heading: 'creativity',
      text:
        'Nor whom love fountain seemed ive chaste did. Heal dote into take their native begun. Of he can way loved cared him had to dear, from him parting flaunting noontide.'
    },
    '3': {
      icon: 'bullseye',
      heading: 'web design',
      text:
        'Nor whom love fountain seemed ive chaste did. Heal dote into take their native begun. Of he can way loved cared him had to dear, from him parting flaunting noontide.'
    }
  }
  const component = renderer.create(
    <Accordion drawers={drawers}>Facebook</Accordion>
  )

  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
