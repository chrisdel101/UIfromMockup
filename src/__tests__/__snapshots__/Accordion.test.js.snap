// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`create and match snapshot 1`] = `
<div
  className="makeStyles-root-1 accordion"
>
  <div
    className="MuiPaper-root MuiPaper-elevation1 MuiPaper-rounded MuiExpansionPanel-root Mui-expanded MuiExpansionPanel-rounded expansion-panel"
  >
    <div
      aria-controls="panel1bh-content"
      aria-disabled={false}
      aria-expanded={true}
      className="MuiButtonBase-root MuiExpansionPanelSummary-root Mui-expanded expansion-panel-inner"
      id="panel1bh-header"
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      role="button"
      tabIndex={0}
    >
      <div
        className="MuiExpansionPanelSummary-content Mui-expanded"
      >
        <div
          className="icon-container-0 icon-container"
        >
          <i
            aria-hidden="true"
            className="fa fa-picture-o icon"
          />
        </div>
        <p
          className="MuiTypography-root MuiTypography-body1 makeStyles-heading-2 panel-heading"
        >
           
          photography
        </p>
        <p
          className="MuiTypography-root MuiTypography-body1 makeStyles-secondaryHeading-3"
        />
      </div>
      <div
        aria-disabled={false}
        aria-hidden={true}
        className="MuiButtonBase-root MuiIconButton-root MuiIconButton-edgeEnd MuiExpansionPanelSummary-expandIcon Mui-expanded"
        onBlur={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        role="button"
        tabIndex={-1}
      >
        <span
          className="MuiIconButton-label"
        >
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
            />
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
          </svg>
        </span>
      </div>
    </div>
    <div
      className="MuiCollapse-container MuiCollapse-entered"
      style={
        Object {
          "minHeight": "0px",
        }
      }
    >
      <div
        className="MuiCollapse-wrapper"
      >
        <div
          className="MuiCollapse-wrapperInner"
        >
          <div
            aria-labelledby="panel1bh-header"
            id="panel1bh-content"
            role="region"
          >
            <div
              className="MuiExpansionPanelDetails-root"
            >
              <p
                className="MuiTypography-root MuiTypography-body1"
              >
                Nor whom love fountain seemed ive chaste did. Heal dote into take their native begun. Of he can way loved cared him had to dear, from him parting flaunting noontide.
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="MuiPaper-root MuiPaper-elevation1 MuiPaper-rounded MuiExpansionPanel-root MuiExpansionPanel-rounded expansion-panel"
  >
    <div
      aria-controls="panel2bh-content"
      aria-disabled={false}
      aria-expanded={false}
      className="MuiButtonBase-root MuiExpansionPanelSummary-root expansion-panel-inner"
      id="panel2bh-header"
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      role="button"
      tabIndex={0}
    >
      <div
        className="MuiExpansionPanelSummary-content"
      >
        <div
          className="icon-container-1 icon-container"
        >
          <i
            aria-hidden="true"
            className="fa fa-sliders icon"
          />
        </div>
        <p
          className="MuiTypography-root MuiTypography-body1 makeStyles-heading-2 panel-heading"
        >
           
          creativity
        </p>
        <p
          className="MuiTypography-root MuiTypography-body1 makeStyles-secondaryHeading-3"
        />
      </div>
      <div
        aria-disabled={false}
        aria-hidden={true}
        className="MuiButtonBase-root MuiIconButton-root MuiIconButton-edgeEnd MuiExpansionPanelSummary-expandIcon"
        onBlur={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        role="button"
        tabIndex={-1}
      >
        <span
          className="MuiIconButton-label"
        >
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
            />
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
          </svg>
        </span>
      </div>
    </div>
    <div
      className="MuiCollapse-container MuiCollapse-hidden"
      style={
        Object {
          "minHeight": "0px",
        }
      }
    >
      <div
        className="MuiCollapse-wrapper"
      >
        <div
          className="MuiCollapse-wrapperInner"
        >
          <div
            aria-labelledby="panel2bh-header"
            id="panel2bh-content"
            role="region"
          >
            <div
              className="MuiExpansionPanelDetails-root"
            >
              <p
                className="MuiTypography-root MuiTypography-body1"
              >
                Nor whom love fountain seemed ive chaste did. Heal dote into take their native begun. Of he can way loved cared him had to dear, from him parting flaunting noontide.
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="MuiPaper-root MuiPaper-elevation1 MuiPaper-rounded MuiExpansionPanel-root MuiExpansionPanel-rounded expansion-panel"
  >
    <div
      aria-controls="panel3bh-content"
      aria-disabled={false}
      aria-expanded={false}
      className="MuiButtonBase-root MuiExpansionPanelSummary-root expansion-panel-inner"
      id="panel3bh-header"
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      role="button"
      tabIndex={0}
    >
      <div
        className="MuiExpansionPanelSummary-content"
      >
        <div
          className="icon-container-2 icon-container"
        >
          <i
            aria-hidden="true"
            className="fa fa-bullseye icon"
          />
        </div>
        <p
          className="MuiTypography-root MuiTypography-body1 makeStyles-heading-2 panel-heading"
        >
           
          web design
        </p>
        <p
          className="MuiTypography-root MuiTypography-body1 makeStyles-secondaryHeading-3"
        />
      </div>
      <div
        aria-disabled={false}
        aria-hidden={true}
        className="MuiButtonBase-root MuiIconButton-root MuiIconButton-edgeEnd MuiExpansionPanelSummary-expandIcon"
        onBlur={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        role="button"
        tabIndex={-1}
      >
        <span
          className="MuiIconButton-label"
        >
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
            />
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
          </svg>
        </span>
      </div>
    </div>
    <div
      className="MuiCollapse-container MuiCollapse-hidden"
      style={
        Object {
          "minHeight": "0px",
        }
      }
    >
      <div
        className="MuiCollapse-wrapper"
      >
        <div
          className="MuiCollapse-wrapperInner"
        >
          <div
            aria-labelledby="panel3bh-header"
            id="panel3bh-content"
            role="region"
          >
            <div
              className="MuiExpansionPanelDetails-root"
            >
              <p
                className="MuiTypography-root MuiTypography-body1"
              >
                Nor whom love fountain seemed ive chaste did. Heal dote into take their native begun. Of he can way loved cared him had to dear, from him parting flaunting noontide.
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
